lib/galaxy/tools/actions/__init__.py

```diff
@@ -359,7 +359,7 @@ def execute(self, tool, trans, incoming=None, return_job=False, set_output_hid=T
         # datasets first, then create the associations
         parent_to_child_pairs = []
         child_dataset_names = set()
-        object_store_populator = ObjectStorePopulator(app)
+        object_store_populator = ObjectStorePopulator(app, user=trans.user)
```

Comment:
Can you open a new PR with these changes - https://github.com/galaxyproject/galaxy/compare/dev...jmchilton:user_objectstore_populator?expand=1. This continues a theme along with https://github.com/galaxyproject/galaxy/pull/10208 and https://github.com/galaxyproject/galaxy/pull/10212 of trying to establish Galaxy abstractions that restrict the code needed to implement this functionality just to object store, quota, and model code.

Reply:
https://github.com/galaxyproject/galaxy/pull/10223